homeassistant/components/eight_sleep/__init__.py

```diff
@@ -113,54 +120,68 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
         # No users, cannot continue
         return False
 
-    hass.data[DOMAIN] = {
-        DATA_API: eight,
-        DATA_HEAT: heat_coordinator,
-        DATA_USER: user_coordinator,
+    dev_reg = async_get(hass)
+    assert eight.device_data
+    device_data = {
+        "manufacturer": "Eight Sleep",
+        "model": eight.device_data.get("modelString", UNDEFINED),
+        "hw_version": eight.device_data.get("sensorInfo", {}).get(
+            "hwRevision", UNDEFINED
+        ),
+        "sw_version": eight.device_data.get("firmwareVersion", UNDEFINED),
```

Comment:
It would be nice to use the constants

`homeassistant/const.py:ATTR_HW_VERSION: Final = "hw_version"`